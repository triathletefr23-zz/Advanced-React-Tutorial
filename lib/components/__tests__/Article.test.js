import React from 'react';
import shallow from '../../../enzyme/setup';
import toJson from 'enzyme-to-json';
import Article from '../Article';

describe('Article', () => {
    const testProps = {
        article: {
            authorId: 123,
            date: "Mon Jul 11 2016 00:00:00 GMT+0000 (UTC)",
            title: "testArticle",
            body: "testtesttesttesttesttesttesttest"
        }
    }

    const author = {
        id: 123,
        firstName: "fn",
        lastName: "ln",
        website: "site.com"
    }

    const testContext = {
        store: {
            lookupAuthor: jest.fn().mockReturnValueOnce(author)
        }
    }

    it("renders correctly to snapshot", () => {
        const wrapper = shallow(
            <Article {...testProps}/>, { context: testContext });
       
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("have correct props", () => {
        const wrapper = shallow(
            <Article {...testProps}/>, { context: testContext });

        expect(wrapper.getElement().props.article).toBe(testProps.article);
        expect(wrapper.getElement().props.author).toBe(testProps.author);
    });

    it("have a store with a function", () => {
        const wrapper = shallow(
            <Article {...testProps}/>, { context: testContext });
        
        expect(wrapper.getElement().props.store.lookupAuthor).toBeCalledWith(author.id);
    });
});
