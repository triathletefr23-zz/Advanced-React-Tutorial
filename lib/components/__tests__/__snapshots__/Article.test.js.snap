// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article renders correctly to snapshot 1`] = `
<Article
  article={
    Object {
      "authorId": 123,
      "body": "testtesttesttesttesttesttesttest",
      "date": "Mon Jul 11 2016 00:00:00 GMT+0000 (UTC)",
      "title": "testArticle",
    }
  }
  author={
    Object {
      "firstName": "fn",
      "id": 123,
      "lastName": "ln",
      "website": "site.com",
    }
  }
  store={
    Object {
      "lookupAuthor": [MockFunction] {
        "calls": Array [
          Array [
            123,
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": Object {
              "firstName": "fn",
              "id": 123,
              "lastName": "ln",
              "website": "site.com",
            },
          },
        ],
      },
    }
  }
/>
`;
